connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/PluginType.java

```diff
@@ -17,22 +17,24 @@
 package org.apache.kafka.connect.runtime.isolation;
 
 import org.apache.kafka.common.config.provider.ConfigProvider;
-import org.apache.kafka.connect.connector.Connector;
 import org.apache.kafka.connect.connector.policy.ConnectorClientConfigOverridePolicy;
 import org.apache.kafka.connect.rest.ConnectRestExtension;
 import org.apache.kafka.connect.sink.SinkConnector;
 import org.apache.kafka.connect.source.SourceConnector;
 import org.apache.kafka.connect.storage.Converter;
+import org.apache.kafka.connect.storage.HeaderConverter;
 import org.apache.kafka.connect.transforms.Transformation;
+import org.apache.kafka.connect.transforms.predicates.Predicate;
 
 import java.util.Locale;
 
 public enum PluginType {
     SOURCE(SourceConnector.class),
     SINK(SinkConnector.class),
-    CONNECTOR(Connector.class),
```

Comment:
This makes me wonder why CONNECTOR was ever in this enum. Do you know @mimaison / @C0urante ?

Reply:
I think it allowed for classes that subclassed `Connector` but didn't subclass from `SinkConnector` or `SourceConnector` to still be detected by the worker during plugin scanning and included in responses for the `/connector-plugins` endpoint, with a type of `UNKNOWN`.

With this change, those kinds of connector classes won't appear in the response for this endpoint anymore.

I think this should be fine. As of https://github.com/apache/kafka/pull/2604, which was merged over a year ago, it's impossible to create that kind of connector anyways.